arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -222,6 +222,11 @@ RestStatus RestCursorHandler::registerQueryOrCursor(VPackSlice const& slice) {
     CursorRepository* cursors = _vocbase.cursorRepository();
     TRI_ASSERT(cursors != nullptr);
     _cursor = cursors->createQueryStream(std::move(query), batchSize, ttl);
+    if (_cursor == nullptr) {   // soft shutdown, no longer allowed
+      generateError(Result(TRI_ERROR_SHUTTING_DOWN,
+                           "coordinator soft shutdown, new cursors blocked"));
+      return RestStatus::DONE;
+    }
```

Comment:
This creates some coupling between the code here and the code in `createQueryStream`, which I think is unnecessary.
I would prefer if `createQueryStream` simply threw an exception in case of soft shutdown, and return a valid pointer in all other cases.
In case it throws, it can already throw the correct error code in the exception, so we wouldn't need to know anything about soft shutdown here.
All exceptions are handled at least on the outermost layer in all RestHandlers automatically, so we can just let any exception from the CursorRepository propagate from here.

Reply:
Good point, will do.

Reply:
Done.